Kernel/KBOComparator.cpp

```diff
@@ -154,8 +170,15 @@ KBOComparator::KBOComparator(TermList tl1, TermList tl2, const KBO& kbo)
   }
 }
 
-bool KBOComparator::check(const SubstApplicator* applicator) const
+bool KBOComparator::check(const SubstApplicator* applicator)
 {
+  if (!_ready) {
```

Comment:
Why do we do this lazy-loading thing instead of calling it in the constructor? Is it sometimes unused?

Reply:
Yes, it might be never used. For example, we create the object when inserting the demodulator into the LHS index, but only initialize it when we try to demodulate with it. Of course, the initialization itself could be also done lazily, i.e. only processing the comparator to the point we actually use it, I just couldn't get around to it this time and it didn't seem like a too hot spot.